app/controllers/comments_controller.rb

```diff
@@ -12,28 +12,36 @@ def create
       track_created(@comment)
     end
 
-    redirect_to polymorphic_path([current_project, @comment.commentable], anchor: dom_id(@comment))
+    redirect_to_comment(@comment)
   end
 
   def update
     if @comment.update_attributes(comment_params)
       track_updated(@comment)
     end
 
-    redirect_to polymorphic_path([current_project, @comment.commentable], anchor: dom_id(@comment))
+    redirect_to_comment(@comment)
   end
 
   def destroy
     if @comment.destroy
       track_destroyed(@comment)
     end
 
-    redirect_to [current_project, @comment.commentable]
+    redirect_to_comment(@comment)
   end
 
   private
 
   def comment_params
     params.require(:comment).permit(:content, :commentable_type, :commentable_id)
   end
+
+  def redirect_to_comment(comment)
+    if comment.persisted? && !request.referrer.nil?
+      request.env['HTTP_REFERER'] += "##{dom_id(comment)}"
```

Comment:
I want to use an `anchor` with `redirect_back`, and I came up with this.
Better ideas?